js/core/utils/common.js

```diff
@@ -262,58 +262,60 @@ export const grep = function(elements, checkFunction, invert) {
     return result;
 };
 
-const arraysEqualByValue = function(array1, array2, depth) {
+const compareArrays = (array1, array2, depth, options) => {
     if(array1.length !== array2.length) {
         return false;
     }
 
-    for(let i = 0; i < array1.length; i++) {
-        if(!equalByValue(array1[i], array2[i], depth + 1)) {
-            return false;
-        }
-    }
-
-    return true;
+    return !array1.some((item, idx) => !compareByValue(item, array2[idx], depth + 1, {
+        ...options,
+        strict: true,
+    }));
 };
 
-const objectsEqualByValue = function(object1, object2, depth, strict) {
-    for(const propertyName in object1) {
-        if(
-            Object.prototype.hasOwnProperty.call(object1, propertyName) &&
-            !equalByValue(object1[propertyName], object2[propertyName], depth + 1, strict)
-        ) {
-            return false;
-        }
-    }
+const compareObjects = (object1, object2, depth, options) => {
+    const keys1 = Object.keys(object1);
+    const keys2 = Object.keys(object2);
 
-    for(const propertyName in object2) {
-        if(!(propertyName in object1)) {
-            return false;
-        }
+    if(keys1.length !== keys2.length) {
+        return false;
     }
 
-    return true;
+    return !keys1.some((key) =>
+        !Object.prototype.hasOwnProperty.call(object2, key)
```

Comment:
According to [doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys), `Object.keys` already returns only own properties, so we can remove `hasOwnProperty`

Reply:
Yup, I agree with you
Let's remove it